src/bundler/bundle_v2.zig

```diff
@@ -9581,13 +9577,19 @@ pub const LinkerContext = struct {
         if (chunk.content == .javascript) {
             const sources = c.parse_graph.input_files.items(.source);
             for (chunk.content.javascript.parts_in_chunk_in_order) |part_range| {
-                const source: Logger.Source = sources[part_range.source_index.get()];
+                const source: *Logger.Source = &sources[part_range.source_index.get()];
 
                 const file_path = brk: {
                     if (source.path.isFile()) {
                         // Use the pretty path as the file name since it should be platform-
                         // independent (relative paths and the "/" path separator)
+                        const path = source.path;
+                        if (path.pretty.ptr == path.text.ptr) {
+                            const rel = bun.path.relativePlatform(c.resolver.fs.top_level_dir, path.text, .loose, false);
```

Comment:
probably should use `pathWithPrettyInitialized` to handle an edgecase in dev server

Reply:
I dont think I have a BundleV2 ref to use here

Reply:
Ok centralized this logic in one place now, should handle the dev server edgecase.